lnwallet/chainfee/estimator.go

```diff
@@ -628,9 +628,9 @@ var _ Estimator = (*BitcoindEstimator)(nil)
 // implementation of this interface in order to allow the WebAPIEstimator to
 // be fully generic in its logic.
 type WebAPIFeeSource interface {
-	// GetFeeMap will query the web API, parse the response and return a
+	// GetResponse will query the web API, parse the response and return a
 	// map of confirmation targets to sat/kw fees.
-	GetFeeMap() (map[uint32]uint32, error)
+	GetResponse() (*WebAPIResponse, error)
```

Comment:
This is about as generic of a name as it could be, is there any way we can specify the contents of the response itself? It's not totally a disaster given that it has a concrete type and is bound to an interface but it would still be better if we named it something else. FWIW I think the old name was probably closer to what we want.

Secondly, I might recommend we don't use a pointer here and instead use a value. There's no need to do heap allocation, is there?

Reply:
cool updated as suggested.